library/Rules/DateTimeDiff.php

```diff
@@ -0,0 +1,152 @@
+            throw new InvalidRuleConstructorException(
+                '"%s" is not a valid type of age (Available: %s)',
+                $this->type,
+                ['y', 'm', 'd', 'days', 'h', 'i', 's', 'f']
+            );
+        }
+        $this->rule = $this->extractSiblingSuitableRule(
+            $rule,
+            new InvalidRuleConstructorException('DateTimeDiff must contain exactly one rule')
+        );
+    }
+
+    public function evaluate(mixed $input): Result
+    {
+        $compareTo = $this->createDateTimeObject($input);
+        if ($compareTo === null) {
+            return Result::failed($input, $this);
+        }
+
+        $dateTimeResult = $this->bindEvaluate(
+            binded: new DateTime($this->format), 
+            binder: $this, 
+            input: $input
+        );
+        if (!$dateTimeResult->isValid) {
+            return $dateTimeResult;
+        }
+
+        $now = $this->now ?? new DateTimeImmutable();
+
+        $nextSibling = $this->rule
+            ->evaluate($this->comparisonValue($now, $compareTo))
+            ->withNameIfMissing($input instanceof DateTimeInterface ? $input->format('c') : $input);
+
+        $parameters = [
+            'type' => $this->getTranslatedType($this->type), 
+            'now' => $this->nowParameter($now)
+        ];
+
+        return (new Result($nextSibling->isValid, $input, $this, $parameters))->withNextSibling($nextSibling);
+    }
+
+    private function comparisonValue(DateTimeInterface $now, DateTimeInterface $compareTo)
+    {
+        return $compareTo->diff($now)->{$this->type};
+    }
+
+    private function nowParameter(DateTimeInterface $now): string
+    {
+        if ($this->format === null && $this->now === null) {
+            return 'now';
+        }
+
+        if ($this->format === null) {
+            return $now->format('Y-m-d H:i:s.u');
+        }
+
+        return $now->format($this->format);
+    }
+
+    private function createDateTimeObject(mixed $input): ?DateTimeInterface
+    {
+        if ($input instanceof DateTimeInterface) {
+            return $input;
+        }
+
+        if (!is_scalar($input)) {
+            return null;
+        }
+
+        if ($this->format === null) {
+            return new DateTimeImmutable((string) $input);
+        }
+
+        $format = $this->getExceptionalFormats()[$this->format] ?? $this->format;
+        $dateTime = DateTimeImmutable::createFromFormat($format, (string) $input);
+        if ($dateTime === false) {
+            return null;
+        }
+
+        return $dateTime;
+    }
+
+    private function getTranslatedType(string $type): string
+    {
+        return match ($type) {
+            'y' => 'years',
+            'm' => 'months',
+            'd' => 'days',
+            'days' => 'full days',
+            'h' => 'hours',
+            'i' => 'minutes',
+            's' => 'seconds',
+            'f' => 'microseconds',
+        };
+    }
+}
```

Comment:
If we are going to use this, I believe that we can improve to a Enum file. I didn't create because the project don't have any